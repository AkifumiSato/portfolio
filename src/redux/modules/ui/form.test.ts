import * as snapshotDiff from 'snapshot-diff'
import reducer, {
  FormModel,
  changeCommentAction,
  changeEmailAction,
  changeNameAction,
} from './form'

const initialState = new FormModel()

test('[form ui state]: init', () => {
  expect(
    snapshotDiff(undefined, reducer(undefined, { type: '@@INIT' }))
  ).toMatchSnapshot()
})

test('[form ui state]: name change', () => {
  expect(
    snapshotDiff(initialState, reducer(initialState, changeNameAction()))
  ).toMatchSnapshot()
})

test('[form ui state]: email change', () => {
  expect(
    snapshotDiff(initialState, reducer(initialState, changeEmailAction()))
  ).toMatchSnapshot()
})

test('[form ui state]: comment change', () => {
  expect(
    snapshotDiff(initialState, reducer(initialState, changeCommentAction()))
  ).toMatchSnapshot()
})
