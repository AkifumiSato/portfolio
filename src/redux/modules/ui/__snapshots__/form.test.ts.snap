// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[form ui state]: comment change 1`] = `
"Snapshot Diff:
- First value
+ Second value

  Immutable.Record {
    \\"isChangeName\\": false,
    \\"isChangeEmail\\": false,
-   \\"isChangeComment\\": false,
+   \\"isChangeComment\\": true,
  }"
`;

exports[`[form ui state]: email change 1`] = `
"Snapshot Diff:
- First value
+ Second value

  Immutable.Record {
    \\"isChangeName\\": false,
-   \\"isChangeEmail\\": false,
+   \\"isChangeEmail\\": true,
    \\"isChangeComment\\": false,
  }"
`;

exports[`[form ui state]: init 1`] = `
"Snapshot Diff:
  Comparing two different types of values. Expected undefined but received object."
`;

exports[`[form ui state]: name change 1`] = `
"Snapshot Diff:
- First value
+ Second value

  Immutable.Record {
-   \\"isChangeName\\": false,
+   \\"isChangeName\\": true,
    \\"isChangeEmail\\": false,
    \\"isChangeComment\\": false,
  }"
`;
